warehouse/query-core/src/main/java/datawave/query/jexl/visitors/PatternNodeVisitor.java

```diff
@@ -34,4 +36,30 @@ public Object visit(ASTERNode node, Object data) {
         
         return patterns;
     }
+    
+    // Ensure we short circuit these nodes
```

Comment:
Is this not exactly what the ShortCircuitBaseVisitor does for you?  Why have these methods here?

Reply:
No, the ShortCircuitBaseVisitor does not include these specific functions. It originally contained them, but it was discovered that other visitors required specific logic in those methods so it was easier to pull them out and implement them for each extending visitor.